flutter-gui-tests/testSrc/io/flutter/tests/gui/InspectorTest.kt

```diff
@@ -6,94 +6,163 @@
+      step("Details selection synced with main tree") {
+        inspectorTree.selectRow(2, expand = true)
+        expect("[[root], MyApp, MaterialApp, MyHomePage]") { inspectorTree.selectionSync().toString() }
+        expect("[MyHomePage]") { detailsTree.selectionSync().toString() }
+        inspectorTree.selectRow(10, expand = true)
+        expect("[[root], MyApp, MaterialApp, MyHomePage, Scaffold, FloatingActionButton]") {
+          inspectorTree.selectionSync().toString()
+        }
+        val string = detailsTree.selectionSync().toString()
+        expect(true) {
+          string.startsWith("[MyHomePage,") && string.endsWith("FloatingActionButton]")
+        }
+      }
+
+      // This is disabled due to an issue in the test framework. The #selectRow call causes
+      // the widget tree to change its selection, which is absolutely not what we want.
+      //      step("Details selection leaves main tree unchanged") {
+      //        val string = detailsTree.selectionSync().toString()
+      //        detailsTree.selectRow(1, expand = false)
+      //        pause(object : Condition("Details tree changes") {
+      //          override fun test(): Boolean {
+      //            return string != detailsTree.selectionSync().toString()
+      //          }
+      //        }, Timeouts.seconds05)
+      //        expect("[[root], MyApp, MaterialApp, MyHomePage, Scaffold, FloatingActionButton]") {
+      //          inspectorTree.selectionSync().toString()
+      //        }
+      //      }
+
       runner().stop()
     }
   }
 
-  private fun IdeFrameFixture.runner(): ExecutionToolWindowFixture.ContentFixture {
-    return runToolWindow.findContent("main.dart")
+  @Test
+  fun hotReload() {
+    ProjectCreator.importProject()
+    ideFrame {
+      launchFlutterApp()
+      val inspector = flutterInspectorFixture(this)
+      inspector.populate()
+      val widgets = inspector.widgetsFixture()
+      val widgetsTree = widgets.inspectorTreeFixture(isDetails = false)
+      widgetsTree.selectRow(0)
+      val detailsTree = widgets.inspectorTreeFixture(isDetails = true)
+      val initialDetails = detailsTree.selectionSync().toString()
+
+      editor {
+        // Wait until current file has appeared in current editor and set focus to editor.
+        moveTo(0)
+        val editorCode = getCurrentFileContents(false)!!
+        val original = "pushed the button this"
+        val index = editorCode.indexOf(original) + original.length
+        moveTo(index)
+        val key = if (isMac) KeyEvent.VK_BACK_SPACE else KeyEvent.VK_DELETE
+        for (n in 1..4) typeKey(key)
+        typeText("that")
+        //typeKey(KeyEvent.VK_ESCAPE) // Dismiss completion popup -- not needed with "that" but is needed with "so"
+      }
+
+      step("Trigger Hot Reload and wait for it to finish") {
+        val reload = findHotReloadButton()
+        reload.click()
+        editor.clickCenter() // Need to cycle the event loop to get button enabled on Mac.
```

Comment:
I bet this was fun to figure out. 😬

Reply:
Oh yes, as someone with an extensive background in window testing would understand :)